.circleci/config.yml

```diff
@@ -31,6 +31,19 @@ yarn_install: &yarn_install
   name: Install Dependencies
   command: yarn install --frozen-lockfile --non-interactive
 
+codecov_install: &codecov_install
```

Comment:
Great for adding the codecov for showing code coverage after each PR changed and tested 👍 

Maybe you could rename the `Install codecov` to `Upload the code coverage data to codecov` to prevent confusion that you show the install codecov, but the command is to [upload the code coverage](https://docs.codecov.com/docs/codecov-uploader).

NIT: If possible, may we could use circleci  [codecov orb](https://circleci.com/developer/orbs/orb/codecov/codecov) which like GitHub Action, for making it simple

Reply:
> Great for adding the codecov for showing code coverage after each PR changed and tested 👍
>
> Maybe you could rename the Install codecov to Upload the code coverage data to codecov to prevent confusion that you show the install codecov, but the command is to [upload the code coverage](https://docs.codecov.com/docs/codecov-uploader).

The script here downloads the binary  file from codecov and check the signature, It doesn't upload any data~ We upload data in the last step of test job: `- run: ./codecov -t ${CODECOV_TOKEN}`, I'll add some comment here.

> NIT: If possible, maybe we could use circleci [codecov orb](https://circleci.com/developer/orbs/orb/codecov/codecov) which like GitHub Action, for making it simple

To use 3rd party orb we need to update our organization security policy, it might affect other projects of Canner, so I download the file manually.

https://app.circleci.com/pipelines/github/Canner/vulcan/53/workflows/eb92a347-0a49-4e3e-bb56-a097d77bbd53/jobs/76

Reply:
Ohoh, got it!
It my misunderstanding, thanks for the explanation, also thanks for telling me the reason why not use the 3rd party and provide tested link too 😺